File: content/en/api/downtimes/code_snippets/api-monitor-update-downtime.rb
require 'dogapi'

api_key = '<DATADOG_API_KEY>'
app_key = '<DATADOG_APPLICATION_KEY>'

dog = Dogapi::Client.new(api_key, app_key)

# Update a downtime
downtime_id = 4336
dog.update_downtime(
  downtime_id,
  scope: 'env:testing',
  end: Time.now.to_i + 60_000,
  message: 'Doing some testing on staging.'
)

File: content/en/api/downtimes/code_snippets/api-monitor-update-downtime.py
from datadog import initialize, api
import time

options = {
    'api_key': '<DATADOG_API_KEY>',
    'app_key': '<DATADOG_APPLICATION_KEY>'
}

initialize(**options)

# Get an existing downtime
stagingDowntimes = []
downtimes = api.Downtime.get_all()

for item in downtimes:
    if item['scope'] == ['env:staging']:
        stagingDowntimes.append(item)

# Update that downtime
api.Downtime.update(
    stagingDowntimes[0]['id'],
    scope='env:staging',
    end=int(time.time()) + 60000,
    message="Doing some testing on staging."
)
